x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/RuntimeUtils.java

```diff
@@ -54,8 +96,10 @@ static void logSearchResponse(SearchResponse response, Logger logger) {
             aggsNames.append(aggs.get(i).getName() + (i + 1 == aggs.size() ? "" : ", "));
         }
 
+        SearchHit[] hits = response.getHits().getHits();
+        int count = hits != null ? hits.length : 0;
         logger.trace("Got search response [hits {}, {} aggregations: [{}], {} failed shards, {} skipped shards, "
-                + "{} successful shards, {} total shards, took {}, timed out [{}]]", response.getHits().getTotalHits(), aggs.size(),
+                + "{} successful shards, {} total shards, took {}, timed out [{}]]", count, aggs.size(),
```

Comment:
Why `getTotalHits()` wasn't the best option?

Reply:
It requires accurate counts - if total hits is disabled (don't track hits), it returns null. Plus it does not tell you the number of results returned but rather the total number of results found.